tests/e2e/ctl_v3_auth_cluster_test.go

```diff
@@ -0,0 +1,130 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.etcd.io/etcd/tests/v3/framework/config"
+	"go.etcd.io/etcd/tests/v3/framework/e2e"
+	"sync"
+	"testing"
+)
+
+func TestAuthCluster(t *testing.T) {
+	e2e.BeforeTest(t)
+	cfg := &e2e.EtcdProcessClusterConfig{
+		ClusterSize:   1,
+		InitialToken:  "new",
+		SnapshotCount: 2,
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	epc, err := e2e.NewEtcdProcessCluster(ctx, t, cfg)
+	if err != nil {
+		t.Fatalf("could not start etcd process cluster (%v)", err)
+	}
+	defer func() {
+		if err := epc.Close(); err != nil {
+			t.Fatalf("could not close test cluster (%v)", err)
+		}
+	}()
+
+	epcClient := epc.Client()
+	createUsers(ctx, t, epcClient)
+
+	if err := epcClient.AuthEnable(ctx); err != nil {
+		t.Fatalf("could not enable Auth: (%v)", err)
+	}
+
+	numKeys := 10
+	testUserClientOpts := e2e.WithAuth("test", "testPassword")
+	rootUserClientOpts := e2e.WithAuth("root", "rootPassword")
+	writeKeys(ctx, epc.Client(testUserClientOpts), numKeys, "test")
+
+	if err := epc.StartNewProc(ctx, t, rootUserClientOpts); err != nil {
+		t.Fatalf("could not start second etcd process (%v)", err)
+	}
+
+	writeKeys(ctx, epc.Client(testUserClientOpts), numKeys, "test_two_nodes")
+
+	hashKvs, err := epc.Client(rootUserClientOpts).HashKV(ctx, int64(numKeys*2))
+	if err != nil {
+		t.Fatalf("could not HashKV (%v)", err)
+	}
+
+	revisionSet := false
+	var revision int64
+	var hash uint32
+	for _, v := range hashKvs {
+		if !revisionSet {
+			revision = v.Header.GetRevision()
+			hash = v.Hash
+			revisionSet = true
+			continue
+		}
+		if revision != v.Header.GetRevision() || hash != v.Hash {
+			t.Errorf("Inconsistent revisions found:")
+			for _, v := range hashKvs {
+				t.Errorf("%+v", v)
+			}
+			t.Fail()
+		}
+	}
```

Comment:
It's over complicated to me. I think we just need to create a new client connecting to the new member, and confirm that it can read & write key successfully.